Release Process + Term Refinement

CooperStansbury opened the pull request:
1. Updated terms (to a hopefully stable point) based on several discussions with @zachll and @grosscol.
1. Ran `tools/cpo_release.sh` to control `dictionary.md` + manage ontology version annotation
1. Fixed bonkers annotation properties.

CooperStansbury committed:
refactored terms and annotation properties
standardized annotations and citations
slotted classes to new homes in BFO
strip obsolete classes from DICTIONARY

based on conversations with ZLL and CG on 08/16/2018

CooperStansbury committed:
build iterations of cpo_release.sh
include local release logging for robot reports
automate dictionary updates

CooperStansbury committed:
imported RO - full instead of RO core
to give ontology greater expressivity between
CPO classes

CooperStansbury committed:
add temp demonstrative causal_pathways
to show Zach how we might format

CooperStansbury committed:
updated terms to reflect discussion with ZLL
core classes changed to ICEs in order to
clearly distinguish between the spek and reality

grosscol reviewed (commented):
I have no comments for the added/removed terms.

The causal_pathways.json is a mock up of the general structure of a json-ld representation of a causal pathway?

What is the purpose of the release_log and release_txt?  These look like intermediate artifacts from the release process.  Why commit them to the repo?

CooperStansbury commented:
1. Yes - we'll discuss the json.ld file tomorrow. I just needed a place to put it. 
1. My thought was a file is slightly more accessible than commit history. I'm open to throwing report messaging to stdout instead of logging file.

CooperStansbury committed:
add spellcheck to release process
require user input to finish cpo_release
measures to improve usability of process

grosscol commented:
Looks good.  Squash?

grosscol commented on tools/cpo_release.sh:
```diff
@@ -39,6 +39,9 @@ robot report --input ${CPO}
 ${SCRIPT_DIR}/make-dict.sh
 
+# spell-check dictionary post release
+aspell check ${SCRIPT_DIR}/DICTIONARY.md
```
You're going to want to add a check that aspell is available in a similar way that robot is checked for.

CooperStansbury replied:
I know, I'm working on this now...

CooperStansbury commented:
@grosscol made some changes to the release process, including compatibility checking for `aspell`.

`cpo_release.sh` would `rm` the file if there were failures, and this isn't good. 

please review and I will happily squash once you sign off. 

Thanks!

grosscol commented on tools/cpo_release.sh:
```diff
@@ -0,0 +1,66 @@
+while true; do
+    read -p "Are you ready to release this version?" yn
+    case $yn in
+        [Yy]* ) rm cpo.owl; mv cpo-updated.owl cpo.owl; rm catalog-v001.xml; echo 'RELEASE COMPLETE.'; break;;
```
Why remove the catalog file?  Xml files currently ignored by git (.gitignore) so it won't end up in the repo anyway.

CooperStansbury replied:
Just for the local system. I find myself removing this file a lot.

grosscol replied:
Interesting.  I would not expect that the catalog needs to be removed and recreated frequently.  If removing it becomes a problem for other users, we can fix it then.

grosscol reviewed (commented):
This looks fine.  Only one question about removing the  catalog.

The user prompt at the end is overkill, but not a bad kind of overkill. Git reset provides the same effect.

CooperStansbury commented:
@grosscol squashed. please merge!
